packages/utils/sets-utils/src/cell-set-utils.js

```diff
@@ -509,30 +509,44 @@ export function treeToCellPolygonsBySetNames(
  * Given a tree with state, get the sizes of the
  * sets currently marked as "visible".
  * @param {object} currTree A tree object.
+ * @param {array} allNamePaths Array of all paths.
  * @param {array} selectedNamePaths Array of arrays of strings,
- * representing set "paths".
+ * representing selected paths.
  * @param {object[]} setColor Array of objects with the
  * properties `path` and `color`.
  * @param {string} theme "light" or "dark" for the vitessce theme
  * @returns {object[]} Array of objects
  * with the properties `name`, `size`, `key`,
  * and `color`.
  */
-export function treeToSetSizesBySetNames(currTree, selectedNamePaths, setColor, theme) {
+export function treeToSetSizesBySetNames(
+  currTree, allNamePaths, selectedNamePaths, setColor, theme,
+) {
   const sizes = [];
-  selectedNamePaths.forEach((setNamePath) => {
-    const node = treeFindNodeByNamePath(currTree, setNamePath);
+
+  const contains = (path, paths) => paths.some((p) => {
+    if (p.length !== path.length) return false;
+    return p.every((value, index) => value === path[index]);
+  });
+
+  allNamePaths.forEach((clusterPath) => {
+    const node = treeFindNodeByNamePath(currTree, clusterPath);
     if (node) {
       const nodeSet = nodeToSet(node);
-      const nodeColor = setColor?.find(d => isEqual(d.path, setNamePath))?.color
-        || getDefaultColor(theme);
-      sizes.push({
+      const nodeColor = setColor?.find(d => isEqual(d.path, clusterPath))?.color
+          || getDefaultColor(theme);
+      const nodeProps = {
         key: generateKey(),
         name: node.name,
         size: nodeSet.length,
         color: nodeColor,
-        setNamePath,
-      });
+        setNamePath: clusterPath,
+        shown: 0,
+      };
+      if (contains(clusterPath, selectedNamePaths)) {
+        nodeProps.shown = 1;
```

Comment:
```suggestion
        shown: false,
      };
      if (contains(clusterPath, selectedNamePaths)) {
        nodeProps.shown = true;
```

Can you add a comment about what the `shown` property is intended to be used for?

Reply:
I have added a comment and renamed the variable to "isGrayedOut", hoping that now it is easier to understand what the code does. What do you think -- is it better?

Reply:
Yes i think that is easy to understand